Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -122,34 +125,82 @@ class BackForwardListViewController: UIViewController, UITableViewDataSource, UI
         guard let bvc = self.bvc else {
             return
         }
-        tableView.snp_updateConstraints { make in
-            make.bottom.equalTo(self.view).offset(bvc.shouldShowFooterForTraitCollection(newCollection) ? -UIConstants.ToolbarHeight : 0)
+        if bvc.shouldShowFooterForTraitCollection(newCollection) != snappedToBottom {
+            self.dismissing = true
```

Comment:
What's the need for the `dismissing` boolean? Were you seeing timing issues betwene `willTransitionToTraitCollection` and `viewWillTransitionToSize`?